jvm/buildSrc/src/main/java/com/radixdlt/cargo/desktop/CargoDesktopPlugin.kt

```diff
@@ -46,35 +48,31 @@ class CargoDesktopPlugin : Plugin<Project> {
                             workingDir = projectDir.parentFile.parentFile
                             commandLine(
                                 "cp",
-                                "target/${triple.rustTargetTripleName}/${buildType.lowercase}/${triple.binaryName}",
-                                "${projectDir}/src/main/resources/${triple.jnaName}/${triple.binaryName}"
+                                "target/${current.rustTargetTripleName}/${buildType.lowercase}/${current.binaryName}",
+                                "${buildDir}/generated/src/resources/${current.jnaName}/${current.binaryName}"
                             )
                         }
                     }
                 }
             }
+        }
+    }
+}
 
-            val cleanTask = tasks.register("cargoClean") {
-                group = BasePlugin.BUILD_GROUP
-
-                doLast {
-                    DesktopTargetTriple.supported.forEach { triple ->
-                        exec {
-                            workingDir = rootDir.parentFile
-                            println("Cleaning for ${triple.rustTargetTripleName}")
-                            commandLine("cargo", "clean", "--target", triple.rustTargetTripleName)
-                        }
-                    }
-                }
-            }
-
-            tasks.getByName("compileJava") {
-                dependsOn(cargoTask)
-            }
-
-            tasks.getByName("clean") {
-                dependsOn(cleanTask)
-            }
+fun Project.currentTargetTriple(): DesktopTargetTriple {
```

Comment:
Helper function to let gradle detect the underlying target triple by running `rustc --version --verbose` and reading from `host:`